docs/boot-process.md

```diff
@@ -0,0 +1,382 @@
+# Bottlerocket Boot Process
+
+This document describes how Bottlerocket boots, focusing on the systemd target progression and service dependencies.
+
+**Keywords:** boot, systemd, targets, preconfigured, configured, multi-user, fipscheck, drivers, sysinit, services, dependencies, ordering, API system, bootstrap containers, settings, startup, initialization, kernel modules
+
+## Overview
+
+Bottlerocket's boot sequence progresses through six main stages, each represented by a systemd target:
+
+```
+sysinit.target
+    ↓
+fipscheck.target (FIPS mode only)
```

Comment:
This isn't the correct order, but the reason why is subtle. There's a `bootconfig-fips.conf` snippet in `release` that sets the default systemd target to `fipscheck.target`.

Since all the units that are needed by that target are `DefaultDependencies=no`, `sysinit.target` doesn't end up added to the job queue until we reach `activate-preconfigured.target`.